src/main/java/org/zeromq/ZMsg.java

```diff
@@ -396,55 +398,65 @@ public void dump(Appendable out)
         catch (IOException e) {
             throw new RuntimeException("Message dump exception " + super.toString(), e);
         }
+        return this;
     }
 
-    public void dump()
+    public ZMsg dump()
     {
         dump(System.out);
+        return this;
     }
 
     // ********* Convenience Deque methods for common data types *** //
 
-    public void addFirst(String stringValue)
+    public ZMsg addFirst(String stringValue)
     {
         addFirst(new ZFrame(stringValue));
+        return this;
     }
 
-    public void addFirst(byte[] data)
+    public ZMsg addFirst(byte[] data)
     {
         addFirst(new ZFrame(data));
+        return this;
     }
 
-    public void addLast(String stringValue)
+    public ZMsg addLast(String stringValue)
     {
         addLast(new ZFrame(stringValue));
+        return this;
     }
 
-    public void addLast(byte[] data)
+    public ZMsg addLast(byte[] data)
     {
         addLast(new ZFrame(data));
+        return this;
     }
 
     // ********* Convenience Queue methods for common data types *** //
 
-    public void push(String str)
+    public ZMsg push(String str)
     {
         push(new ZFrame(str));
+        return this;
     }
 
-    public void push(byte[] data)
+    public ZMsg push(byte[] data)
     {
         push(new ZFrame(data));
+        return this;
     }
 
-    public boolean add(String stringValue)
```

Comment:
Seeing it a bit late, but this breaks API, as the return value is not consistent anymore.
We may consider keeping the signature and adding a new method for the chaining call. I just do not know which name to use: and(), plus(), ...

Reply:
I saw this boolean is always true, so I thought it's redundant and I can remove it. for the name of the new method I recommend addAnd() because two methods with the names add() and plus() in one class, could be a cause of misleading